lib/engine/hex.rb

```diff
@@ -60,16 +60,46 @@ def name
     end
 
     def lay(tile)
+      # key: city on @tile (AKA old_city)
+      # values: city on tile (AKA new_city)
+      #
+      # map old cities to new based on edges they are connected to
+      city_map =
+        # if @tile is blank, map cities by index
+        if @tile.cities.flat_map(&:connected_edges).empty? && (@tile.cities.size == tile.cities.size)
+          @tile.cities.zip(tile.cities).to_h
+
+        # if @tile is not blank, ensure connectivity is maintained
+        else
+          @tile.cities.map do |old_city|
+            cities = tile.cities.select do |new_city|
+              # we want old_edges to be subset of new_edges
+              old_edges = old_city.connected_edges
+              new_edges = new_city.connected_edges
+              old_edges & new_edges == old_edges
+            end.compact
+
+            unless cities.one?
```

Comment:
it it possible that it maps to another city?

Reply:
Probably not...got rid of this check in cc2a5d3